src/fragmentarium/domain/Genres.ts

```diff
@@ -0,0 +1,90 @@
+import produce, { castDraft, Draft, immerable } from 'immer'
+import _ from 'lodash'
+
+export class Genre {
+  [immerable] = true
+  readonly category: ReadonlyArray<string>
+  readonly uncertain: boolean
+
+  constructor(category: string[], uncertain: boolean) {
+    this.category = category
+    this.uncertain = uncertain
+  }
+
+  setUncertain(uncertain: boolean): Genre {
+    return produce(this, (draft: Draft<Genre>) => {
+      draft.uncertain = uncertain
+    })
+  }
+
+  get toString(): string {
+    return this.category.join('-').replace(' ', '_')
+  }
+}
+
+export class Genres {
+  [immerable] = true
+  readonly genres: ReadonlyArray<Genre>
+
+  constructor(genres: Genre[]) {
+    this.genres = genres
+  }
+
+  static fromJSON(
```

Comment:
Use camel case for function names (i.e. `fromJson`). That would be consistent with the rest of our code.